va/validation-authority.go

```diff
@@ -93,20 +125,16 @@ func (va ValidationAuthorityImpl) validateDvsni(identifier core.AcmeIdentifier,
 	RS := append(R, S...)
 
 	z := sha256.Sum256(RS)
-	zName := fmt.Sprintf("%x.acme.invalid", z)
+	zName := fmt.Sprintf("%064x.acme.invalid", z)
 
 	// Make a connection with SNI = nonceName
 
-	hostPort := ""
+	hostPort := identifier.Value + ":443"
 	if va.TestMode {
 		hostPort = "localhost:5001"
-	} else {
-		if identifier.Type != "dns" {
-			challenge.Status = core.StatusInvalid
-			return
-		}
-		hostPort = identifier.Value + ":443"
 	}
+	va.log.Notice(fmt.Sprintf("Attempting to validate DVSNI for %s %s %s",
+		identifier, hostPort, zName))
 	conn, err := tls.Dial("tcp", hostPort, &tls.Config{
```

Comment:
This should probably have a `defer conn.Close()` statement after the error catch or a file handle for the connection will be left hanging around.